readthedocs/proxito/views/mixins.py

```diff
@@ -30,52 +36,129 @@ class ServeDocsMixin:
 
     version_type = INTERNAL
 
-    def _serve_docs(
-            self,
-            request,
-            final_project,
-            path,
+    def _serve_docs(self, request, project, version, filename, check_if_exists=False):
+        """
+        Serve a documentation file.
+
+        :param check_if_exists: If `True` we check if the file exists before trying
+         to serve it.
```

Comment:
What are the cases where it's useful one of the other? It would be good to expand this docstring explaining that.